### PR TITLE
Improve logging statuser

### PR DESCRIPTION
**Pull request by lzap**
When I was searching in logs today I noticed that the statuser do not have account information for each log. Also it is missing any correlation id. So this patch adds that, also, when I was on that I also add source_id too so we can easily search for availability checks per source id.

In a separate patch, I noticed a typo in a variable name. Example log session:

```
9:25AM TRC Received message with key: 1, topic: platform.provisioning.internal.availability-check, offset: 0, partition: 0 binary=pbstatuser hostname=mone.home.lan payload="{\"source_id\":\"1\"}" version=HEAD
9:25AM TRC Getting authentication from source 1 account_number=6395343 binary=pbstatuser client=sources hostname=mone.home.lan org_id=13446659 source_id=1 trace_id=52fdfc072182654f163f5f0f9a621d72 version=HEAD
9:25AM WRN Could not get authentication: cannot list source authentication: Get "http://nuc:8131/api/sources/v3.1/sources/1/authentications": dial tcp 192.168.1.2:8131: connect: connection refused account_number=6395343 binary=pbstatuser hostname=mone.home.lan org_id=13446659 source_id=1 trace_id=52fdfc072182654f163f5f0f9a621d72 version=HEAD
9:25AM TRC Received message with key: 1, topic: platform.provisioning.internal.availability-check, offset: 1, partition: 0 binary=pbstatuser hostname=mone.home.lan payload="{\"source_id\":\"1\"}" version=HEAD
9:25AM TRC Getting authentication from source 1 account_number=6395343 binary=pbstatuser client=sources hostname=mone.home.lan org_id=13446659 source_id=1 trace_id=9566c74d10037c4d7bbb0407d1e2c649 version=HEAD
9:25AM WRN Could not get authentication: cannot list source authentication: Get "http://nuc:8131/api/sources/v3.1/sources/1/authentications": dial tcp 192.168.1.2:8131: connect: connection refused account_number=6395343 binary=pbstatuser hostname=mone.home.lan org_id=13446659 source_id=1 trace_id=9566c74d10037c4d7bbb0407d1e2c649 version=HEAD
9:25AM DBG looking up offset of kafka reader for partition 0 of platform.provisioning.internal.availability-check: 2 binary=pbstatuser hostname=mone.home.lan kafka=true version=HEAD
```

### CODE REVIEW

File: cmd/pbstatuser/main.go [hunk 1]
```diff
@@ -39,7 +40,7 @@ const ChannelBuffer = 32
 type SourceInfo struct {
 	Authentication clients.Authentication
 
-	SourceApplictionID string
+	SourceApplicationID string
```
Comment (lzap) [inline cmd/pbstatuser/main.go:43]: This is the typo.

File: cmd/pbstatuser/main.go [hunk 2]
```diff
@@ -66,6 +67,10 @@ func processMessage(ctx context.Context, message *kafka.GenericMessage) {
 
 	sourceId := asm.SourceID
 
+	// Set source id as logging field
+	logger = ptr.To(logger.With().Str("source_id", sourceId).Logger())
+	ctx := ctxval.WithLogger(origCtx, logger)
```
Comment (lzap) [inline cmd/pbstatuser/main.go:72]: Either this or introducing a temporary variable so I could do `logger = &tempVar`

File: internal/kafka/kafka.go [hunk 3]
```diff
@@ -200,10 +202,26 @@ func (b *kafkaBroker) Consume(ctx context.Context, topic string, since time.Time
...
+			if !traceId.IsValid() {
+				traceId = random.TraceID()
+			}
+			ctx = ctxval.WithTraceId(ctx, traceId.String())
+
+			newLogger := logger.With().
+				Str("trace_id", traceId.String()).
+				Str("account_number", identity.Identity.AccountNumber).
+				Str("org_id", identity.Identity.OrgID).Logger()
+			ctx = ctxval.WithLogger(ctx, &newLogger)
```
Comment (lzap) [inline internal/kafka/kafka.go:223]: We do the same for REST API in `trace_id.go` middleware.

**Review by adiabramovitch — APPROVED** [review]
Looks good 
TY @lzap

**Pull request merged**